File: leetcode/js/how-many-numbers-are-smaller-than-the-current-number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function (nums) {
  const result = [];

  for (const num of nums) {
    const count = nums.filter((n) => n < num).length;
    result.push(count);
  }

  return result;
};

var smallerNumbersThanCurrent = function (nums) {
  const sorted = [...nums].sort((a, b) => a - b);
  const countMap = sorted.reduce((acc, cur, idx) => {
    if (!acc.has(cur)) {
      acc.set(cur, idx);
    }

    return acc;
  }, new Map());

  return nums.map((num) => countMap.get(num));
};

File: leetcode/rust/how-many-numbers-are-smaller-than-the-current-number.rs
impl Solution {
    pub fn smaller_numbers_than_current(nums: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::new();

        for num in nums.iter() {
            let count = nums.iter().filter(|&n| n < num).count();
            result.push(count as i32);
        }

        result
    }
}
